src/main/java/de/adesso/anki/battle/world/bodies/Mine.java

```diff
@@ -30,9 +45,37 @@ public void evaluateBehavior() {
 
 	@Override
 	public void evaluateBehavior(MqttService mqtt) throws MqttException {
-		// TODO Auto-generated method stub
-
+		World world = this.getWorld();
+		if ( checkCollision(this,world)) {
+			world.getBodiesModifiable().remove(this);
+		}
 	}
+		private boolean checkCollision(Body weapon, World world) {
+			
+			//merge into weapon superclass
+	    	if (weapon instanceof Mine && !((Mine) weapon).isActive()) {
+	    		return false;
+	    	}
+
+	    	List<Vehicle> vehicles = world.getVehicles();
+			Position pos1 = weapon.getPosition();
+			boolean destroy = false;
+			//TODO find damage values for weapon types

```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)